src/hotspot/cpu/riscv/riscv.ad

```diff
@@ -8172,6 +8249,116 @@ instruct convF2HF_reg_reg(iRegINoSp dst, fRegF src, fRegF ftmp, iRegINoSp xtmp)
   ins_pipe(pipe_slow);
 %}
 
+// half precision operations
+
+instruct reinterpretS2HF(fRegF dst, iRegINoSp src)
+%{
+  match(Set dst (ReinterpretS2HF src));
+  effect(TEMP_DEF dst);
```

Comment:
I don't see why this `TEMP_DEF dst` effect is needed. Maybe we should remove it from all the newly-added instructs.

Reply:
OK.